piccolo/query/methods/aggregate.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import annotations
+
+import typing as t
+
+from piccolo.columns import Column, Selectable
+
+NUMERIC_TYPES = (
+    "BIGINT",
+    "INTEGER",
+    "NUMERIC",
+    "REAL",
+    "SMALLINT",
+)
+
+
+class Avg(Selectable):
+    """
+    AVG() SQL function. The column type must be numeric to run the query.
+
+    await Band.select(Avg(Band.popularity)).run()
+    """
+
+    def __init__(self, column=t.Type[Column]):
+        self.column = column
+
+    def get_select_string(self, engine_type: str, just_alias=False) -> str:
+        if self.column.column_type.split("(")[0] in NUMERIC_TYPES:
+            column_name = self.column._meta.get_full_name(
+                just_alias=just_alias
+            )
```

Comment:
An alternative to this is to check the column's `value_type`.

```python
if self.column.value_type in (int, decimal.Decimal, float):
    column_name = self.column._meta.get_full_name(
        just_alias=just_alias
    )
```

It's slightly more flexible, as this way people can define their own column types and it will still work. What do you think?

It's probably worth moving the logic into a separate method called something like `is_number_column`, as it reduces the code repetition, and makes unit testing easier.

```python
def is_number_column(column: Column) -> bool:
    return column.value_type in (int, decimal.Decimal, float)
```

Reply:
Also, if we move the check to the `__init__` method it might be better, as otherwise it will just detect the problem when the query runs. What do you think?

Reply:
@dantownsend Thank you. These are great suggestions, I will try to implement it and write more tests.